pages/DSNP/Operations.md

```diff
@@ -0,0 +1,36 @@
+
+Where operations are listed as using control keys or ownership proofs as input parameters, this indicates that they should be provided in addition to those providing invocation authentication.
+
+## Transaction Identifiers
+
+Each invocation of a DSNP Operation should be associated with a transaction identifier. The precise data type and usage pattern of a transaction identifier is left implementation-dependent.
+Transaction identifiers are useful for association of operations with asynchronously emitted state change records. It MUST be possible to associate a DSNP State Change Record with a transaction identifier from a particular DSNP Operation invocation.
+
+## Failure Handling
+
+Compliant implementations may respond to error conditions either synchronously, as a response to the invocation request, or asynchronously, by emitting a [Failure Record](Records.md#failure).
+
+## List of Operations
+
+| Operation | Optional? | Principal(s) | Inputs | State Change Record |
+|---------- |---------- |------------- |------- |-------------- |
+| <a id="create-identifier">Create Identifier</a> | no | None | Control Key, Control Key Ownership Proof | [Identifier Creation Record](Records.md#identifier-creation) |
+| <a id="retire-identifier">Retire Identifier</a> | no | User | None | [Identifier Retirement Record](Records.md#identifier-retirement) |
+| <a id="define-delegation">Define Delegation</a> | no | User AND Delegate | User's Identifier, Delegate's Identifier, Set of Allowed [Announcement Types](Announcements.md#announcement-types) | [Delegation Definition Record](Records.md#delegation-definition) |
+| <a id="revoke-delegation">Revoke Delegation</a> | no | User OR Delegate | User's Identifier, Delegate's Identifier | [Delegation Revocation Record](Records.md#delegation-revocation) |
+| <a id="add-control-key">Add Control Key</a> | YES | User | Control Key, Control Key Ownership Proof | [Control Key Addition Record](Records.md#control-key-addition) |
+| <a id="remove-control-key">Remove Control Key</a> | YES | User | Control Key | [Control Key Removal Record](Records.md#control-key-removal) |
+| <a id="publish-announcement">Publish Announcement</a> | no[^1] | User OR Delegate | [Announcement](Announcements.md) | [Announcement Published Record](Records.md#announcement-published) |
+| <a id="publish-batch">Publish Batch</a> | no[^1] | User OR Delegate | [Announcement Type](Announcements.md#announcement-types), [Batch Publication](BatchPublications.md) URL, Batch Publication Content Hash | [Batch Published Record](Records.md#batch-published) |
+
+[^1]: For each Announcement Type, an implementation may support one or both of these operations.
+    Implementations MUST document which of the operations is available for each announcement type.
```

Comment:
Syntax: Odd characters on screen after this sentence.

Reply:
It looks like this syntax for footnotes is not completely standard for markdown tools. I'm going to change it to an asterisk instead.